src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.SingleSymbolRenameSession.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Rename.ConflictEngine
+{
+    internal static partial class ConflictResolver
+    {
+        private class SingleSymbolRenameSession : Session
+        {
+            private readonly string _originalText;
+            private readonly string _replacementText;
+
+            public static async Task<SingleSymbolRenameSession> CreateAsync(
+                SymbolicRenameLocations renameLocationSet,
+                Location renameSymbolDeclarationLocation,
+                string replacementText,
+                ImmutableArray<SymbolKey> nonConflictSymbolKeys,
+                CodeCleanupOptionsProvider fallbackOptions,
+                CancellationToken cancellationToken)
+            {
+                var originalText = renameLocationSet.Symbol.Name;
+                var renameLocations = renameLocationSet.Locations;
+                var symbol = renameLocationSet.Symbol;
+
+                // only process documents which possibly contain the identifiers.
+                var documentIdOfRenameSymbolDeclaration = renameLocationSet.Solution.GetRequiredDocument(renameSymbolDeclarationLocation.SourceTree!).Id;
+
+                using var _ = PooledDictionary<DocumentId, DocumentRenameInfo>.GetInstance(out var documentIdToRenameInfoBuilder);
+                var (documentsIdsToBeCheckedForConflict, possibleNameConflicts) = await FindDocumentsAndPossibleNameConflictsAsync(
+                    renameLocationSet,
+                    replacementText,
+                    originalText,
+                    cancellationToken).ConfigureAwait(false);
+
+                var replacementTextValid = IsIdentifierValid_Worker(
```

Comment:
Can we move `IsIdentifierValid_Worker` to this class? I don't think it's called elsewhere

Reply:
Yeah I can put it here in this PR, but in my next PR I need to move it back to the base type again : )